sw/host/tests/usbdev/usbdev_stream/stream_test.cc

```diff
@@ -0,0 +1,574 @@
+// Copyright lowRISC contributors (OpenTitan project).
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+//
+// USB streaming data test
+//
+// Linux host-side application that receives a stream of LFSR-generated data
+// from the USB device, checks the received bytestream and then XORs it with a
+// host-side LFSR-generated byte stream to transmit back to the device.
+//
+// By default the streaming test expects a number of USB serial connections to
+// the target device, one port per endpoint:
+//
+//   /dev/ttyUSB0 - supplies and receives LFSR-generated byte stream for one/
+//                  the only endpoint
+//   /dev/ttyUSB1 - a secondary stream
+//   /dev/ttyUSB..
+//
+// Note that the mapping from device endpoints to USB port number is not
+// guaranteed, and  when multiple streams are used, it is _not_ necessarily the
+// case that ascending streams/endpoints in usbdev_stream_test are mapped to
+// a contiguous range of ascending ttyUSBi port names.
+//
+// Either or both of the initial input port and the initial output port may be
+// overridden using command line parameters.
+//
+// Usage:
```

Comment:
I don't know if it is necessary to repeat the usage here given that there is the usage below in the code

Reply:
I think I'll leave it. It was more useful before, granted, because it started as a single-file program. I like being able to look at the top of the source to a simple program and see the command line used to invoke/build it. Of course, over a period of some months of verification effort it has become quite a bit more involved.